thewalrus/_hafnian.py

```diff
@@ -819,21 +771,23 @@ def hafnian(
     if loop:
         return loop_hafnian(A, D=None, reps=None, glynn=glynn)
 
-    return _haf(A, reps=None, glynn=glynn)
+    if recursive:
+        if loop:
+            raise TypeError("Recursive algorithm does not support the loop hafnian")
```

Comment:
Best to use a warning here as `raise` will halt execution.
```suggestion
            warnings.warn("Recursive algorithm does not support the loop hafnian")
```

Don't forget to add `import warnings` at the top of the file.